component/config.go

```diff
@@ -27,13 +27,8 @@ type Config any
 var configValidatorType = reflect.TypeOf((*ConfigValidator)(nil)).Elem()
 
 // UnmarshalConfig helper function to UnmarshalConfig a Config.
-// It checks if the config implements confmap.Unmarshaler and uses that if available,
-// otherwise uses Map.UnmarshalExact, erroring if a field is nonexistent.
+// Deprecated: [v0.101.0] Use conf.Unmarshal(&intoCfg)
 func UnmarshalConfig(conf *confmap.Conf, intoCfg Config) error {
```

Comment:
Do we want to change the way this function works with the deprecation? Is there a reason we cannot leave the function unchanged until it's completely removed?

Reply:
It does, it removes the old way to call into unmarshaling by calling the Unmarshaler Unmarshal function.

Reply:
AIUI this is a no-op change